op/opstatus.go

```diff
@@ -140,6 +143,35 @@ func (s *Status) Check() HealthResult {
 	return hr
 }
 
+// AddCheckerWithMetrics adds a function that can check the applications health and meter the outcomes of the checks
+// Multiple checkers are allowed.  The checker functions should be capable of
+// being called concurrently (with each other and with themselves).
+func (s *Status) AddCheckerWithMetrics(name string, checkerFunc func(cr *CheckResponse)) *Status {
+	checkCounterVec := prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: safeMetricName(name),
+		Help: "Counts the number of times the healthcheck was run based on the outcome of the check",
+	}, []string{"healthcheck_result"})
+
+	s.checkers = append(s.checkers, checker{name, checkerFunc, checkCounterVec})
+	return s
+}
+
+func safeMetricName(checkName string) string {
```

Comment:
What is this supposed to do? If it's supposed to replace non safe characters with `_`, it's not doing that currently. So maybe add:
```
else {
        x = x + string(b)    
}
```

Reply:
Good catch.